main.py

```diff
@@ -223,6 +228,12 @@ def post(self):
             self.response.write(json.dumps(ERR_EMAIL_ALREADY_REGISTERED))
             return
 
+        id = convert_email_to_id(email)
```

Comment:
You will also want to check that a schedule exists for this id here, to avoid creating an account with no schedule.